spec/lang/representation.md

```diff
@@ -3,11 +3,17 @@
 The main purpose of [types](types.md) is to define how [values](values.md) are (de)serialized into/from memory.
 This is the *[representation relation]*, which is defined in the following.
 `decode` converts a list of bytes into a value; this operation can fail if the byte list is not a valid encoding for the given type.
-`encode` inverts `decode`; it will always work when the value is [well-formed][well-formed-value] for the given type (which the specification must ensure, i.e. violating this property is a spec bug).
-Unsized types cannot be represented as values and thus a call to `encode` or `decode` for such types can never occur in a well-formed program.
+Some types also have additional runtime constraints to be [valid][valid-value], which is always checked after decoding.
+`encode` inverts `decode`; it, however, will always work when the value is [valid][valid-value] for the given type.
+
+`encode` and `decode` must satisfy some [properties](#generic-properties), which the specification must ensure, i.e. violating these properties is a spec bug.
+However, they can also make some assumptions:
+The types in the representation relation are all [well-formed](well-formed.md#well-formed-layouts-and-types) and sized, as unsized types cannot be represented as values.
+`encode` can also assume the value is [valid][valid-value] for the type,
+and `decode` can assume the size of the byte list matches the size of the type.
 
 [representation relation]: https://github.com/rust-lang/unsafe-code-guidelines/blob/master/reference/src/glossary.md#representation-relation
-[well-formed-value]: well-formed.md#well-formed-values
+[valid-value]: #Value-validity
```

Comment:
I am not sure we want to rename "well-formed" to "valid"... I am trying to move away from the term "validity invariant" (see https://github.com/rust-lang/unsafe-code-guidelines/issues/539), so it seems a step backwards for MiniRust to start using this term now.